NKL/Python.lean

```diff
@@ -44,8 +43,9 @@ inductive Expr where
   deriving Repr
 
 inductive Expr' where
-  | const (value: Const)
-  | name (id: String) (ctx : Ctx)
+  | const (value : Const)
+  | tensor (shape : List Expr) (dtype : String)
```

Comment:
Reserving judgment until reading more, but not obvious to me how this will be used. Maybe some documentation would be good.

Reply:
Yes, this may need to change as more kernels are supported in the tracer. So far, I am able to get by with this basic type information.